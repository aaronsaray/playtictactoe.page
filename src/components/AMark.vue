<template>
  <div
    @mouseover="iAmConsideringThis()"
    @mouseleave="iStoppedConsideringThis()"
    @click="iChoseThis()"
  >
    <span
      v-if="hasMark == 'x' || (mark == 'x' && considering)"
      class="x"
      :class="{ considering: considering }"
    >
      &times;
    </span>
    <span
      v-if="hasMark == 'o' || (mark == 'o' && considering)"
      class="o"
      :class="{ considering: considering }"
    >
      o
    </span>
  </div>
</template>

<script>
export default {
  props: ["hasMark", "mark", "enabled"],

  data: function() {
    return {
      considering: false
    };
  },

  methods: {
    iAmConsideringThis() {
      if (this.enabled && !this.hasMark) {
        this.considering = true;
      }
    },

    iStoppedConsideringThis() {
      this.considering = false;
    },

    iChoseThis() {
      if (this.enabled && !this.hasMark) {
        this.$emit("chose");
        this.considering = false;
      }
    }
  }
};
</script>
