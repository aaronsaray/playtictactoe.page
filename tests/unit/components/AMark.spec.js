import { shallowMount } from "@vue/test-utils";
import AMark from "@/components/AMark.vue";
import { directive } from "@babel/types";

describe("AMark.vue", () => {
  it("renders no marks when nothing is indicated", () => {
    const wrapper = shallowMount(AMark, {
      propsData: {
        hasMark: false,
        mark: "o",
        enabled: true
      }
    });
    expect(wrapper.text()).toMatchSnapshot();
  });

  it.todo("has mark x shows x, no considering");
  it.todo("has mark o shows o, no considering");

  it.todo("is mark x and considering shows x with considering");
  it.todo("is mark o and considering shows o with considering");

  it("will not mark considering if not enabled", () => {
    const wrapper = shallowMount(AMark, {
      propsData: {
        hasMark: false,
        mark: "o",
        enabled: false
      }
    });

    const box = wrapper.find("div");
    box.trigger("mouseover");
    expect(wrapper.vm.considering).toBeFalsy();
  });

  it("will not mark considering if enabled but has mark", () => {
    const wrapper = shallowMount(AMark, {
      propsData: {
        hasMark: "o",
        mark: "o",
        enabled: true
      }
    });

    const box = wrapper.find("div");
    box.trigger("mouseover");
    expect(wrapper.vm.considering).toBeFalsy();
  });

  it("will mark considering if enabled and has no mark", () => {
    const wrapper = shallowMount(AMark, {
      propsData: {
        hasMark: null,
        mark: "o",
        enabled: true
      }
    });

    const box = wrapper.find("div");
    box.trigger("mouseover");
    expect(wrapper.vm.considering).toBeTruthy();
  });

  it("clears considering when mouse leave", () => {
    const wrapper = shallowMount(AMark, {
      propsData: {
        hasMark: null,
        mark: "o",
        enabled: true
      }
    });
    wrapper.setData({ considering: true });

    const box = wrapper.find("div");
    box.trigger("mouseleave");
    expect(wrapper.vm.considering).toBeFalsy();
  });

  it.todo("will not emit that it was chose if not enabled");
  it.todo("will not emit that it was chose if enabled but has a mark");
  it.todo(
    "will emit that it was chose, clear considering if enabled and no mark"
  );
});
