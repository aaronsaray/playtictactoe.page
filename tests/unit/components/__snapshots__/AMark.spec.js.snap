// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AMark.vue renders no marks when nothing is indicated 1`] = `""`;
